airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -72,8 +72,9 @@ public JobCreationOutput createNewJob(final JobCreationInput input) {
       failNonTerminalJobs(input.getConnectionId());
 
       final StandardSync standardSync = configRepository.getStandardSync(input.getConnectionId());
-      if (input.isReset()) {
+      final List<StreamDescriptor> streamsToReset = streamResetPersistence.getStreamResets(input.getConnectionId());
 
+      if (!streamsToReset.isEmpty()) {
```

Comment:
This is now where we decide if a job should be a reset or not - based purely on what is in the stream_reset table. So, in the normal case or in the recovering from unexpected state case, we will still respect what is in the table and run a reset when it is required.